### PR TITLE
add healthchecks

### PR DESCRIPTION
**Pull request by ermirizio**
Add health-checks for all docker containers in compose to remove sleeps

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5245/merge
Commit: 6760fce18bfa0178764b022c7dbb99a16d37d0bf 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5417519544)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5245/merge
Commit: d1e8faee8087a59b1be8e3197559532f43c28b23 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5447756619)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5245/merge
Commit: 86bec48642ac3a9783b2aa900da6ff217bfab2f0 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5453298324)

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5245/merge
Commit: 1ff26ba41c18f1226b8e742b3d9291ae1be0dc44 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5457429709)

File: ci/tests/api-functionality/docker-compose.yml [hunk 1]
```diff
@@ -16,7 +22,28 @@ services:
...
       - TYK_LOGLEVEL=debug
 
+  gw-checker:
+    image: curlimages/curl
+    networks: [ test-api-functionality ]
+    command: tail -f /dev/null
+    depends_on:
+      - gw
+    healthcheck:
+      test: curl -s --fail http://gw:8080/hello | grep -o '.*{"redis":{"status":"pass",'
```
Comment (titpetric) [inline ci/tests/api-functionality/docker-compose.yml:32]: The check with grep could fail based on order of keys in the json objects. Asserting the value with jq will be more reliable but the base image needs to provide jq.

This image is ~5mb and provides curl and jq: https://hub.docker.com/r/badouralix/curl-jq

Comment (gwotg) [reply ci/tests/api-functionality/docker-compose.yml:32]: https://tyktech.atlassian.net/browse/TD-2072

**Review by gwotg — APPROVED** [review]
Thanks @ermirizio ! Besides the comment from Tit which would be addressed on TD-2072, LGTM! Approving to unblock EEE

**Pull request merged**